### PR TITLE
v4 API: Subscribers

### PR DESCRIPTION
**Pull request by n7studios**
## Summary

Adds support for endpoints in the subscribers section of the v4 API.

## Testing

Tests pass.

## Checklist

* [x] I have [written a test](TESTING.md#writing-a-test) and included it in this PR
* [x] I have [run all tests](TESTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](TESTING.md#run-php-codesniffer)
* [x] The code passes when [running PHPStan](TESTING.md#run-phpstan)
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

### CODE REVIEW

File: src/ConvertKit_API.php [hunk 1]
```diff
@@ -1023,56 +1163,64 @@ public function update_subscriber(
...
      */
     public function unsubscribe(string $email)
     {
-        return $this->put(
-            'unsubscribe',
-            ['email' => $email]
+        return $this->post(
+            sprintf(
+                'subscribers/%s/unsubscribe',
+                $this->get_subscriber_id($email)
```
Comment (noelherrick) [inline src/ConvertKit_API.php:1179]: Should this method check to see if this this returns anything, printing a different message and/or returning a different error code if the a subscriber with this email does not exist?

**Review by noelherrick — APPROVED** [review]
Looks good. As far as the lines that cleanup the subscribers (by unsubscribing them), is there a way to put that into a method or block that always runs, even if there's an assertion failure?

File: src/ConvertKit_API.php (hunk 1 above)
Comment (n7studios) [reply src/ConvertKit_API.php:1179]: It throws a `ClientException` with a 404 not found (per test `testUnsubscribeWithNotSubscribedEmailAddress`), which I think is sufficient for the consumer/developer to then handle accordingly.  We have the option for the consumer/developer to both catch exceptions and inspect the [HTTP response code](https://github.com/ConvertKit/ConvertKitSDK-PHP/pull/78), which I think is sufficient here.

Given the v4 API only permits unsubscribing by subscriber ID (`unsubscribe_by_id`), this `unsubscribe` method is more a convenience retained to minimise breaking changes from v3.

At some point I'd like to start throwing our own exceptions to provide improved context within `ClientException`, which you suggested [here](https://github.com/ConvertKit/ConvertKitSDK-PHP/pull/45#issuecomment-1468164236) - but I think that merits its own PR.

**Comment by n7studios** [comment]
> Looks good. As far as the lines that cleanup the subscribers (by unsubscribing them), is there a way to put that into a method or block that always runs, even if there's an assertion failure?

Yes - I'll open a PR for that today.

File: src/ConvertKit_API.php (hunk 1 above)
Comment (noelherrick) [reply src/ConvertKit_API.php:1179]: That makes sense!

**Pull request merged**